src/main/resources/templates/activity_lecturer.html

```diff
@@ -141,15 +141,15 @@ <h6 class="align-top mb-3 border-bottom border-light mr-3 pb-3">Komentarz:</h6>
             </th>
         </tr>
         <thead style="border-top: 3px solid #343a40;">
-        <tr th:if="${#strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER')}">
+        <tr th:if="${#strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER') and user.getCoursesLectured().contains(course)}">
```

Comment:
I don't think strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER') check is necessary - we initialize coursesLectured with an empty collection, so we don't need to worry about null pointer exceptions while checking with user.getCoursesLectured().contains(course). Thus, we can reduce this whole line to just a simple contains check.

Reply:
Yup, good point, I just tacked it on the end without much thinking.